File: DSA_2215/search/hash_table/hash_table_search.go
package main

import "fmt"

type HashTable struct {
	table map[int]int
}

func NewHashTable() *HashTable {
	return &HashTable{
		table: make(map[int]int),
	}
}

func (h *HashTable) Insert(key, value int) {
	h.table[key] = value
}

func (h *HashTable) Search(key int) int {
	if val, found := h.table[key]; found {
		return val
	}
	return -1
}

func useMe() {
	ht := NewHashTable()
	ht.Insert(1, 10)
	ht.Insert(2, 20)
	targetKey := 2
	value := ht.Search(targetKey)
	if value != -1 {
		fmt.Printf("Value for key %d is %d\n", targetKey, value)
	} else {
		fmt.Printf("Key %d not found in the hash table\n", targetKey)
	}
}

func main() {
	useMe()
}

File: DSA_2215/search/hash_table/hash_table_search.dart
class HashTable {
  Map<int, int> _table = {};

  void insert(int key, int value) {
    _table[key] = value;
  }

  int? search(int key) {
    return _table[key];
  }
}

void useMe() {
  HashTable ht = HashTable();
  ht.insert(1, 10);
  ht.insert(2, 20);
  int targetKey = 2;
  int? value = ht.search(targetKey);
  if (value != null) {
    print("Value for key $targetKey is $value");
  } else {
    print("Key $targetKey not found in the hash table");
  }
}

void main() {
  useMe();
}
